src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.cpp

```diff
@@ -90,7 +90,7 @@ bool ChaseMovementGenerator::Update(Unit* owner, uint32 diff)
 
     // our target might have gone away
     Unit* const target = GetTarget();
-    if (!target || !target->IsInWorld())
+    if (!target || !target->IsInWorld() || !target->IsAlive())
```

Comment:
I really struggle to understand why the IsAlive() check wasn't here before

Reply:
ummm good one, I cant remember exactly, but there is actually a reason

all I can remember right now is:
- unit's death is handled before movementgenerator update
- it is _guaranteed_ that movements are cleared on unit's death
- checking for alive shouldn't be needed...

Also, what if a dev needs a death creature chasing someone? (the odds are same as the earth being actually flat, but who knows right?)

Reply:
The check is for the target, with the old code creatures with ChaseMovementGenerator would stare at the dead body of the target until it was removed from world. Now they go back at doing what they are supposed to do

Reply:
umm then I think the problem is within chase movement comprobation/creation:

- My target dies, combat manager does it's thing.
  - Has my target changed? Yes? Am I chasing that new target? Start chasing
  - Am I now in combat? No? Stop chasing

And more use cases that I dont have the time to fully detail 😢

Reply:
But those checks don't work because when target dies the creature is not engaged anymore and that skips the checks that are supposed to trigger evade or whatever. Just debug it and you'll see how broken the current implementation is

Reply:
I see what you mean... damn